src/VisualStudio/Core/Def/Progression/GraphQueryManager.cs

```diff
@@ -21,162 +25,129 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.Progression
     internal class GraphQueryManager
     {
         private readonly Workspace _workspace;
-        private readonly IAsynchronousOperationListener _asyncListener;
 
         /// <summary>
         /// This gate locks manipulation of <see cref="_trackedQueries"/>.
         /// </summary>
         private readonly object _gate = new();
-        private readonly List<ValueTuple<WeakReference<IGraphContext>, List<IGraphQuery>>> _trackedQueries = new();
+        private ImmutableArray<(WeakReference<IGraphContext> context, ImmutableArray<IGraphQuery> queries)> _trackedQueries = ImmutableArray<(WeakReference<IGraphContext>, ImmutableArray<IGraphQuery>)>.Empty;
 
-        // We update all of our tracked queries when this delay elapses.
-        private ResettableDelay? _delay;
+        private readonly AsyncBatchingWorkQueue _updateQueue;
```

Comment:
moved to a simple update-queue model that updates all live-queries after any workspace chnages.